x-pack/plugins/security_solution/public/plugin_services.ts

```diff
@@ -112,6 +112,13 @@ export class PluginServices {
     });
     const customDataService = this.startCustomDataService(query, startPlugins.data);
 
+    const timelineDataService: DataPublicPluginStart = {
+      ...startPlugins.data,
+      query,
+    };
+
+    timelineDataService.query.filterManager = new FilterManager(coreStart.uiSettings);
```

Comment:
Consider extracting this logic into a separate private function (e.g. `createTimelineDataService`), similar to the `customDataService` creation in the line above.

Reply:
Thanks! But now I miss the `new FilterManager`, where has it gone?

Reply:
@semd It's not required actually, since `query` already instantiates a new `FilterManager`. And it turned out, that setting the `filterManager` prop resulted in some broken callbacks, causing this issue: https://github.com/elastic/kibana/pull/180850#issuecomment-2059215963